releasenotes/notes/fix-cpp-for-loop-refs-5a6bb879bf58a4fb.yaml

```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fixes bug in C++ code where certain for-loops were iterating over copies
```

Comment:
I'd not consider this a bug, more an improvement :)
I'd swear that the compiler is able to optimize this to avoid the copy, after all you cannot change a `const` variable, but agree that we shall use references and be explicit about it.